rlinf/runners/agent_runner.py

```diff
@@ -197,13 +202,13 @@ def run(self):
                         logging.info(
                             f"Step limit given by max_steps={self.max_steps} reached. Stopping run"
                         )
-                        return
+                        train_end = True
```

Comment:
this is ugly, how about creating a function called `self.stop_and_cleanup`, calling it and add return here?